go-controller/pkg/ovn/egressip.go

```diff
@@ -1614,26 +1622,22 @@ func (e *egressIPController) deleteEgressReroutePolicy(filterOption, egressIPNam
 		{
 			Model: &logicalRouterPolicy,
 			ModelPredicate: func(lrp *nbdb.LogicalRouterPolicy) bool {
+				if len(lrp.Nexthops) != 1 || lrp.Nexthops[0] != gatewayRouterIP {
```

Comment:
I am not sure this is right. Correct me if I am wrong but according to this, if we have an LRP with `nexthops = [item1, item2]` and we want to remove `item2`, we won't with this code, right? The predicate `len(lrp.Nexthops) != 1` will be true and we will return false when retrieving the LRP, and hence not modify it.

Reply:
heh, let me thins about this. In the  'original' code, lines 1624 were:
```
if len(logicalRouterPolicyRes) == 1 &&
len(logicalRouterPolicyRes[0].Nexthops) == 1 &&
logicalRouterPolicyRes[0].Nexthops[0] == gatewayRouterIP { ...
```
so this is aimed to be a translation from that. When you wrote that, did you not meant to be the case, @alexanderConstantinescu  ?

Reply:
I don't think it's correctly translated then. The code needs to do what the doc bloc says. The previous code modified the `Nexthops` fields irrespective of the amount of nexthops defined. When the last remaining nexthop is the `gatewayRouterIP`:  we need to delete the LRP completely. So imagine an LRP with `nexthops = [item1, item2]`, first update comes in and we need to remove `item2`, we thus only mutate the LRP and have the result be `nexthops = [item1]`. Second update comes in and we need to remove `item1`, we now mutate the LRP and remove `item1` and then also delete the LRP completely.

Reply:
Make sense. Thanks for catching this mistake @alexanderConstantinescu . I resolved this on latest PR commit.